tests/v2/validation/rbac/globalrolesv2/namespacedrules_test.go

```diff
@@ -0,0 +1,531 @@
+	require.NoError(ns.T(), err)
+	listPodsAsAdmin, err := ns.client.WranglerContext.Core.Pod().List(readPodsNS, metav1.ListOptions{})
+	require.NoError(ns.T(), err)
+	
+	require.Equal(ns.T(), len(listPodsAsAdmin.Items), len(listPodsAsUser.Items))
+	require.Equal(ns.T(), listPodsAsAdmin.Items, listPodsAsUser.Items)
+
+	log.Info("Verify user cannot list crtbs in the deleted namespace from the namespaced rules")
+	_, err = userClient.WranglerContext.Mgmt.ClusterRoleTemplateBinding().List(readCrtbsNS, metav1.ListOptions{})
+	require.Error(ns.T(), err)
+	k8sError.IsForbidden(err)
+}
+
+func (ns *NamespacedRulesTestSuite) TestDeleteUserDeletesRolebindingsForNamespacedRules() {
+
+	subSession := ns.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a global role with multiple namespacedRules.")
+
+	readPodsNS, err := createProjectAndAddANamespace(ns.client, "custom-sa-ns")
+	require.NoError(ns.T(),err)
+
+	namespacedRules := map[string][]rbacv1.PolicyRule{
+		readPodsNS: {
+			readServiceAccountPolicy,
+		},
+	}
+	createdGlobalRole, err := createGlobalRoleWithNamespacedRules(ns.client, namespacedRules)
+	require.NoError(ns.T(), err)
+
+	log.Info("Create a user with global role standard user and custom global role.")
+	createdUser, err := users.CreateUserWithRole(ns.client, users.UserConfig(), standardUser, createdGlobalRole.Name)
+	require.NoError(ns.T(), err)
+
+	ns.validateNSRulesRBACResources(createdUser, namespacedRules)
+	
+	grbOwner, err := getGlobalRoleBindingForUser(ns.client, createdUser.ID)
+	require.NoError(ns.T(), err)
+	nameSelector := fmt.Sprintf("metadata.name=%s-%s", grbOwner, namespace)
+
+	err = ns.client.WranglerContext.Mgmt.User().Delete(createdUser.ID, &metav1.DeleteOptions{})
+	require.NoError(ns.T(), err)
+
+	rbs, err := rbac.ListRoleBindings(ns.client, localcluster, namespace, metav1.ListOptions{FieldSelector: nameSelector})
+	require.NoError(ns.T(), err)
+	require.Empty(ns.T(), rbs.Items)
+}
+
+func (ns *NamespacedRulesTestSuite) TestWebhookRejectsEmptyVerbsInNamespacedRules() {
+
+	subSession := ns.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a global role with namespacedRules having empty verbs.")
+
+	readSecretsPolicy.Verbs = []string{}
+	namespacedRules := map[string][]rbacv1.PolicyRule{
+		globalDataNamespace: {
+			readSecretsPolicy,
+		},
+	}
+	_, err := createGlobalRoleWithNamespacedRules(ns.client, namespacedRules)
+	require.Error(ns.T(), err)
+	errMessage := "admission webhook \"rancher.cattle.io.globalroles.management.cattle.io\" denied the request: globalrole.namespacedRules." + globalDataNamespace +"[0].verbs: Required value: verbs must contain at least one value"
+
+	require.Equal(ns.T(), errMessage, err.Error())
+}
+
+func (ns *NamespacedRulesTestSuite) TestWebhookIncorrectVerbsNotRejected() {
+
+	subSession := ns.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a global role with namespacedRules having empty verbs.")
```

Comment:
The test case name suggests passing incorrect verb but the log here says 'namespacedRules having empty verbs'

Reply:
Updated it to be `incorrect`